src/core/mavlink_commands.h

```diff
@@ -122,19 +123,20 @@ class MavlinkCommandSender {
         mavlink_message_t mavlink_message{};
         CommandResultCallback callback{};
         dl_time_t time_started{};
+        void* timeout_cookie = nullptr;
 
         explicit Work(double new_timeout_s) : timeout_s(new_timeout_s) {}
     };
 
     void receive_command_ack(mavlink_message_t message);
-    void receive_timeout();
+    void receive_timeout(const uint16_t command);
 
     void call_callback(const CommandResultCallback& callback, Result result, float progress);
 
     SystemImpl& _parent;
     LockedQueue<Work> _work_queue{};
```

Comment:
Can't you now remove that one?

Reply:
It is the one used by `do_work()` to run the commands. When a command is sent, it is now saved in the `unordered_map` which is used by the receiving side. Something like this:

* `_work_queue`: sending queue
* `_sent_commands`: ACK queue

While a command is "active" (i.e. in `_sent_commands`), another command with the same cmd_id cannot be sent by `_work_queue`.

Reply:
Ah ok.